### ISSUE TITLE
Search: Customizer preview when switching post types

### ISSUE THREAD

**Issue by ebinnion** [Automattic/jetpack#8575]
In #8552, I started work on improving the customizer preview. But, I wasn't able to figure out how to either:

- Reload the search results automatically when the user changes the post types for the widget
- Show a notice to the user that the post types had changed and the search would need to be run again

**Comment by gravityrail** [comment]
Just a note here: @ebinnion and I have tried a number of ways to do this, and all have serious flaws. 

Basically the problem is this:

- The widget takes a bunch of its state (search term, sorting, post types) from the URL, via the WP_Query object
- In the customizer, there's no easy way to modify the URL of the preview iframe, as it's tightly managed by the customizer's JS in order to do selective refresh etc.
- In the jetpack-search class, it's impossible to distinguish between which params we are receiving because they're a default, vs user selected, and which might be stale.

We can try passing a widget ID to the WP_Query object, so we can at least know which widget initiated a search and look up the defaults and apply them to the search after its been issued, but we still face the problem of not knowing whether a param is "sorted descending" because it's what the user selected vs. intended to be the default, and there's also the issue of looking up the correct widget configuration from the customizer's wild versioning system.

Basically, this is super hard but I wanted to document why here for future generations...
